### PR TITLE
coreneuron: Added new cmake flags especially for develop version

### PR DESCRIPTION
**Pull request by ohm314**

### CODE REVIEW

File: var/spack/repos/builtin/packages/coreneuron/package.py [hunk 1]
```diff
@@ -154,6 +155,54 @@ def cmake_args(self):
...
+                   '-DCORENRN_ENABLE_OPENMP=%s' % ('ON' if '+openmp' in spec else 'OFF'),
+                   '-DCORENRN_ENABLE_UNIT_TESTS=%s' % ('ON' if '+tests' in spec else 'OFF'),
+                   '-DCORENRN_ENABLE_TIMEOUT=OFF'
+                   ]
+
+        if spec.satisfies('~shared') or spec.satisfies('+gpu'):
+            options.append('-DCOMPILE_LIBRARY_TYPE=STATIC')
+
+        if 'bgq' in spec.architecture and '%xl' in spec:
+            options.append('-DCMAKE_BUILD_WITH_INSTALL_RPATH=1')
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/coreneuron/package.py]:
```suggestion
            options.append('-DCMAKE_BUILD_WITH_INSTALL_RPATH=1')
```

Still in love with BlueGene? ;) 
can be removed now.

File: var/spack/repos/builtin/packages/coreneuron/package.py [hunk 2]
```diff
@@ -154,6 +155,54 @@ def cmake_args(self):
...
+
+        if spec.satisfies('+gpu'):
+            gcc = which("gcc")
+            options.extend(['-DCUDA_HOST_COMPILER=%s' % gcc,
+                            '-DCUDA_PROPAGATE_HOST_FLAGS=OFF',
+                            '-DENABLE_SELECTIVE_GPU_PROFILING=ON',
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/coreneuron/package.py]:
```suggestion
    
```

File: var/spack/repos/builtin/packages/coreneuron/package.py [hunk 3]
```diff
@@ -154,6 +155,54 @@ def cmake_args(self):
...
+                            '-DENABLE_OPENACC=ON',
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/coreneuron/package.py]:
```suggestion
                            '-DCORENRN_ENABLE_GPU =ON'
```

File: var/spack/repos/builtin/packages/coreneuron/package.py [hunk 4]
```diff
@@ -154,6 +155,54 @@ def cmake_args(self):
...
+                            '-DENABLE_OPENACC_INFO=ON'])
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/coreneuron/package.py]:
```suggestion
                            ])
```

File: var/spack/repos/builtin/packages/coreneuron/package.py [hunk 5]
```diff
@@ -154,6 +155,54 @@ def cmake_args(self):
...
+    def get_cmake_args(self):
+        spec   = self.spec
+        flags = self.get_flags()
+
+        if spec.satisfies('+profile'):
+            env['CC']  = 'tau_cc'
+            env['CXX'] = 'tau_cxx'
+        elif 'bgq' in spec.architecture and spec.satisfies('+mpi'):
+            env['CC']  = spec['mpi'].mpicc
+            env['CXX'] = spec['mpi'].mpicxx
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/coreneuron/package.py]: can be removed now

**Review by pramodk — COMMENTED** [review]
lgtm, small changes are missing.

File: var/spack/repos/builtin/packages/coreneuron/package.py (hunk 1 above)
Comment (ohm314) [reply var/spack/repos/builtin/packages/coreneuron/package.py]: I don't understand your suggestion here?

File: var/spack/repos/builtin/packages/coreneuron/package.py (hunk 2 above)
Comment (ohm314) [reply var/spack/repos/builtin/packages/coreneuron/package.py]: This I will only remove from the build for "develop" but leave it for previous versions?

File: var/spack/repos/builtin/packages/coreneuron/package.py [hunk 6]
```diff
@@ -36,7 +36,7 @@ class Coreneuron(CMakePackage):
     homepage = "https://github.com/BlueBrain/CoreNeuron"
     url      = "https://github.com/BlueBrain/CoreNeuron"
 
-    version('develop', git=url, submodules=True)
+    version('develop', git=url, submodules=True, preferred=True)
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/coreneuron/package.py]:
```suggestion
    version('develop', git=url, submodules=True)
```

File: var/spack/repos/builtin/packages/py-bglibpy/package.py [hunk 7]
```diff
@@ -14,6 +14,9 @@ class PyBglibpy(PythonPackage):
 
     version('develop', branch='master')
     version('4.1.4', commit='e54d294460e5bdf6b9990bc10a4606b412b76d90')
+    version('4.1.2', commit='d01c9359edf92e39be5b840b5bb975da1c960988')
+    version('4.0.27', commit='42d9c1f891ef1ec9af6d72c49ff3b7726a009951')
+    version('4.0.17', commit='e90513f52a7d9ca3c16877ae7a2bcb8df31c8545')
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/py-bglibpy/package.py]: this should be removed from here.

**Review by pramodk — COMMENTED** [review]
clean! could you just takeout develop as preferred and remove bglibpy versions?

File: var/spack/repos/builtin/packages/coreneuron/package.py (hunk 1 above)
Comment (pramodk) [reply var/spack/repos/builtin/packages/coreneuron/package.py]: I meant removing all bg-q related stuff (which you have done).

**Review by pramodk — APPROVED** [review]

**Pull request merged**